jar-infer/src/main/java/com/uber/nullaway/jarinfer/DefinitelyDerefedParamsDriver.java

```diff
@@ -30,70 +34,201 @@
+            for (IMethod mtd : Iterator2Iterable.make(cls.getAllMethods().iterator())) {
+              if (!mtd.getDeclaringClass()
+                  .getClassLoader()
+                  .getName()
+                  .toString()
+                  .equals("Primordial")) {
+                // some Application classes are Primordial (why?)
+                Preconditions.checkNotNull(mtd, "method not found");
+                IR ir =
+                    cache
+                        .getIRFactory()
+                        .makeIR(mtd, Everywhere.EVERYWHERE, options.getSSAOptions());
+                ControlFlowGraph<SSAInstruction, ISSABasicBlock> cfg = ir.getControlFlowGraph();
+                Set<Integer> result = new DefinitelyDerefedParams(mtd, ir, cfg, cha).analyze();
+                if (!result.isEmpty()) {
+                  map_mtd_result.put(mtd, result);
+                  map_str_result.put(mtd.getSignature(), result);
+                }
+              }
+            }
+          }
+        }
+      }
+    }
     long end = System.currentTimeMillis();
-    System.out.println("done");
-    System.out.println("took " + (end - start) + "ms");
-    System.out.println("definitely-derefereced paramters: " + result.toString());
+    System.out.println("-----\ndone\ntook " + (end - start) + "ms");
+    System.out.println("definitely-derefereced paramters: " + map_str_result.toString());
 
-    return result;
+    writeJarModel(cha, map_mtd_result);
+    return map_str_result;
   }
 
   /*
-   * Check set equality of results with expected results
+   * Write inferred Jar model in astubx format
    *
    */
-  public boolean verify(Set<String> result, Set<String> expected) {
-    for (String var : result) {
-      if (!expected.remove(var)) return false;
+  private static void writeJarModel(
+      IClassHierarchy cha, HashMap<IMethod, Set<Integer>> map_mtd_result) {
+    try {
+      File aStubXFile = new File(aStubXPath);
+      aStubXFile.createNewFile();
+      DataOutputStream out = new DataOutputStream(new FileOutputStream(aStubXFile, false));
+      Map<String, String> importedAnnotations =
+          new HashMap<String, String>() {
+            {
+              put("@Nonnull", "javax.annotation.Nonnull");
+            }
+          };
+      Map<String, Set<String>> packageAnnotations = new HashMap<>();
+      Map<String, Set<String>> typeAnnotations = new HashMap<>();
+      Map<String, MethodAnnotationsRecord> methodRecords = new LinkedHashMap<>();
+
+      for (Map.Entry<IMethod, Set<Integer>> entry : map_mtd_result.entrySet()) {
+        IMethod mtd = entry.getKey();
+        Set<Integer> ddParams = entry.getValue();
+        if (ddParams.isEmpty()) continue;
+        Map<Integer, ImmutableSet<String>> argAnnotation =
+            new HashMap<Integer, ImmutableSet<String>>();
+        for (Integer param : ddParams) {
+          argAnnotation.put(param, ImmutableSet.of("@Nonnull"));
```

Comment:
No `@` here either. I think it works as long as you are consistent, but elsewhere I am using only the annotation name, not the `@Name` syntax.